### ISSUE TITLE
Loading custom font

### ISSUE THREAD

**Issue by semir-t** [blazer82/FT81x_Arduino_Driver#10]
Is there any example on how to load the custom font using this library ?

**Comment by blazer82** [comment]
No, I haven't implemented this feature. Although it should be possible if you're willing to dig into the FT81x's documentation.

**Comment by semir-t** [comment]
Yes, I am already trying to do that but without any luck. 

I am guided by the following  [application note](https://www.ftdichip.com/Support/Documents/AppNotes/AN_277_FT800_Custom_Font_Implement.pdf ) and I managed to generate the data that needs to be loaded. This data is loaded by following code:

    ft81x.writeGRAM(1000,148,SAMApp_ShowCustomFont_MetricBlock);
    ft81x.writeGRAM(1000 + 148,5510,SAMApp_ShowCustomFont_FontBmpData);
    ft81x.beginDisplayList();
    ft81x.clear(WHITE);
    ft81x.clear(FT81x_COLOR_RGB(255,255,255));
    ft81x.drawText(240, 230, 34, FT81x_COLOR_RGB(0,0,0), FT81x_OPT_CENTER, "Font1\0"); 
    ft81x.sendCommand(BITMAP_HANDLE(6));
    /* ft81x.sendCommand(BITMAP_SOURCE(1000)); */
    /* ft81x.sendCommand(BITMAP_SOURCE(1090)); */
    /* ft81x.sendCommand(BITMAP_SOURCE(-508)); */
    ft81x.sendCommand(BITMAP_SOURCE(-708));
    ft81x.sendCommand(BITMAP_LAYOUT(FT_L1,2,29));
    ft81x.sendCommand(BITMAP_SIZE(FT_NEAREST,FT_BORDER,FT_BORDER,16,29));

    ft81x.sendCommand(0xFFFFFF2BUL);
    ft81x.sendCommand(6);
    ft81x.sendCommand(FT81x_RAM_G + 1000);
    ft81x.sendCommand(0);
    }
    ft81x.drawText(240, 280, 6, FT81x_COLOR_RGB(0,0,0), FT81x_OPT_CENTER, "Font2\0");
    ft81x.swapScreen();`

`
But I only get white screen in a place where text should be and I am running out of ideas. :/

**Comment by blazer82** [comment]
It seems to me that your approach is correct. I haven't tried this myself yet and I'd need to consult the documentation to verify your code, so I can't tell you much more right now.

**Comment by semir-t** [comment]
Ok, I will continue to work on it and will write comment here if I manage to fix it.

**Comment by dBarmpxkos** [comment]
Hello, 
I am trying to move forward with the same task. Any input would be appreciated.
Although I can see the differences in the library code, could you comment on this:
What would be the equivalent using the library to this SPI transaction 
	Ft_Gpu_Hal_WrMemFromFlash(phost, RAM_G + 1000,	MetricBlock,sizeof(MetricBlock));
	Ft_Gpu_CoCmd_Dlstart(phost);

	Ft_App_WrCoCmd_Buffer(phost,BITMAP_HANDLE(7));
	Ft_App_WrCoCmd_Buffer(phost,BITMAP_SOURCE(196));
	Ft_App_WrCoCmd_Buffer(phost,BITMAP_LAYOUT(L8,28,34));
	Ft_App_WrCoCmd_Buffer(phost,BITMAP_SIZE(NEAREST,BORDER,BORDER,28,34));

**Comment by dBarmpxkos** [comment]
Hello,
I am trying to move forward with the same task. Any input would be appreciated.
Although I can see the differences in the library code, could you comment on this:
What would be the equivalent using the library to these SPI transactions. It is related to setting a custom font of some properties, after it is assigned to bitmap handle #7. 

```
Ft_Gpu_Hal_WrMemFromFlash(phost, RAM_G + 1000, MetricBlock,sizeof(MetricBlock));
Ft_Gpu_CoCmd_Dlstart(phost);

Ft_App_WrCoCmd_Buffer(phost,BITMAP_HANDLE(7));
Ft_App_WrCoCmd_Buffer(phost,BITMAP_SOURCE(196));
Ft_App_WrCoCmd_Buffer(phost,BITMAP_LAYOUT(L8,28,34));
Ft_App_WrCoCmd_Buffer(phost,BITMAP_SIZE(NEAREST,BORDER,BORDER,28,34));
			
Ft_App_WrCoCmd_Buffer(phost,DISPLAY());
Ft_Gpu_CoCmd_Swap(phost);

/* Download the commands into fifo */
Ft_App_Flush_Co_Buffer(phost);
```

I was searching for an implementation for the WrCoCmdBuffer command, and found this:
https://github.com/NewhavenDisplay/EVE2-TFT-Modules/blob/d6c02d0663cf49621f987898061493f115dae1e6/4.3in/_480x272/Resistive/FT_App_Graph/Project/Arduino/FT_App_Graph/FT_App_Graph.ino#L95

So, the difference is that this approach writes everything into a FIFO and in the end it sends them to the screen?
I am trying, as semir-t, to use either `sendCommand()` or `write32()` commands, but it looks like they don't do the same thing as in the app note example.

Is there any way I could port this example using your library? I can see that `startCmd()` somehow increases a pointer, but how would I construct the full packet? 
Does any of this make sense? :) I really appreciate your input.
